In [1]:
#dont care who does, as long as it is done. 
#but for for moderator it must be individual

In [85]:
import os
import pandas as pd

In [86]:
# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)
#     print(df)
else:
    print(f"File '{file_name}' does not exist in the specified folder.")

In [87]:
df.head()

Company Name  \
0                         Quickdesk   
1                         QuickDesk   
2                     Public Mutual   
3   RedHill Travel & Tours Sdn Bhd    
4  RESOURCELEAP CONSULTANCY SDN BHD   

                                     Company Address  
0  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
1  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
2  Menara Public Bank 2, No. 78, Jalan Raja Chula...  
3   located at No 36, Tingkat, 1, Jalan Ibrahim, ...  
4  A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...

In [8]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 290 entries, 0 to 289
Data columns (total 2 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Company Name     290 non-null    object
 1   Company Address  290 non-null    object
dtypes: object(2)
memory usage: 4.7+ KB


In [9]:
print('Unique Companies', len(df['Company Name'].unique()))

Unique Companies 254


Of the 290 companies in this xlxs file, only 254 are unique

In [13]:
# Find non-unique companies
non_unique_companies = df['Company Name'].value_counts()[df['Company Name'].value_counts() > 1]

# Print the non-unique companies
print('Non-Unique Companies:')
print(non_unique_companies)

Non-Unique Companies:
Finexus International Sdn Bhd                                    3
Wonder Pod Sdn Bhd                                               3
Infront Consulting Malaysia                                      3
iFAST SERVICE CENTRE SDN BHD                                     3
CTC Global Sdn Bhd                                               3
Kollect Systems Sdn Bhd                                          3
Fusionex                                                         3
Nexagate Sdn Bhd                                                 2
CodeHamster Solution                                             2
SUNNICASIA MALAYSIA SDN BHD                                      2
Speedrent Technology Sdn Bhd                                     2
Al-Noor International School                                     2
Infront Consulting Group (M) Sdn Bhd                             2
Surge Ventures Sdn Bhd                                           2
AIME Healthcare Sdn Bhd                 

In [16]:
# Find non-unique companies
non_unique_companies = df['Company Address'].value_counts()[df['Company Address'].value_counts() > 1]

# Print the non-unique companies
print('Non-Unique Companies Addresses:')
print(non_unique_companies)

Non-Unique Companies Addresses:
Unit 10-02 Block B, Phileo Damansara 1, 9, Jalan 16/11, 46350 Petaling Jaya, Selangor                                 3
WorQ, Unit 3.07, Level 3, KL Gateway Mall Gateway Mall, No 2, Jalan Kerinchi, Pantai Dalam, 59200 Kuala Lumpur        2
Level 12, Tower A, Plaza 33 MSC Cybercenter, 1 Jalan Kemajuan, Section 13, 46200 Petaling Jaya, Selangor, Malaysia    2
Level 21, Menara ExxonMobil, Kuala Lumpur City Centre, 50088, Kuala Lumpur                                            2
4 & 6, Lengkok Bellamy, Bukit Petaling, 50460 Kuala Lumpur, Wilayah Persekutuan Kuala Lumpur                          2
38-2, Jalan Teknologi 3/6C, Kota Damansara, 47810 Petaling Jaya, Selangor                                             2
E-05-01, First Floor\nGarden Shoppe @ One City\nJalan USJ 25/1\n47650 Subang Jaya\nSelangor, Malaysia                 2
Level 3, Menara Axis,No.2,Jalan \n51a/223,Seksyen 51a Petaling Jaya \n46100 Malaysia.                                 2
Level 33

In [12]:
# Find the rows that contain NaN values in the specified columns
nan_rows = df[df.isna().any(axis=1)]

# Print the rows
print(nan_rows)

Empty DataFrame
Columns: [Company Name, Company Address]
Index: []


There are no Nan rows in the dataframe

## Testing out fuzzy matching

In [17]:
pip install fuzzywuzzy


Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [52]:
import os
import pandas as pd
from fuzzywuzzy import fuzz

# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Function to check if two strings are similar using fuzzywuzzy
def is_similar(str1, str2, threshold=95):
    return fuzz.partial_ratio(str1.lower(), str2.lower()) >= threshold

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)

    # Deduplicate similar company names
    similar_companies = {}
    for i, company_name in enumerate(df['Company Name']):
        for name, idx in similar_companies.items():
            if is_similar(company_name, name):
                df.loc[i, 'Company Name'] = name
                break
        else:
            similar_companies[company_name] = i

    # Print the total number of unique companies after deduplication
    unique_companies = df['Company Name'].nunique()
    print('Unique Companies (after deduplication):', unique_companies)

    # Find non-unique companies
    non_unique_companies = df['Company Name'].value_counts()[df['Company Name'].value_counts() > 1]

    # Print the non-unique companies
    print('Non-Unique Companies:')
    print(non_unique_companies)
else:
    print(f"File '{file_name}' does not exist in the specified folder.")


Unique Companies (after deduplication): 214
Non-Unique Companies:
EY                                                                                        7
Telekom Malaysia Berhad                                                                   5
Finexus International Sdn Bhd                                                             5
Petronas Digital Sdn Bhd                                                                  4
Infront Consulting Group (M) Sdn Bhd                                                      4
CTC Global Sdn Bhd                                                                        4
Alnoor International School                                                               4
Centre for Lifelong Education and Learning Innovation (LEARN) at Multimedia University    4
Fusionex                                                                                  4
Infront Consulting Malaysia                                                               3
iFAST SERVICE 

In [24]:
# import os
# import pandas as pd
# from fuzzywuzzy import fuzz

# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Function to check if two strings are similar using fuzzywuzzy
def is_similar(str1, str2, threshold=80):
    return fuzz.partial_ratio(str1.lower(), str2.lower()) >= threshold

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)

    # Deduplicate similar company names
    similar_companies = {}
    for i, company_name in enumerate(df['Company Address']):
        for name, idx in similar_companies.items():
            if is_similar(company_name, name):
                df.loc[i, 'Company Name'] = name
                break
        else:
            similar_companies[company_name] = i

    # Print the total number of unique companies after deduplication
    unique_companies = df['Company Address'].nunique()
    print('Unique Companies (after deduplication):', unique_companies)

    # Find non-unique companies
    non_unique_companies = df['Company Address'].value_counts()[df['Company Address'].value_counts() > 1]

    # Print the non-unique companies
    print('Non-Unique Companies:')
    print(non_unique_companies)
else:
    print(f"File '{file_name}' does not exist in the specified folder.")

Unique Companies (after deduplication): 277
Non-Unique Companies:
Unit 10-02 Block B, Phileo Damansara 1, 9, Jalan 16/11, 46350 Petaling Jaya, Selangor                                 3
WorQ, Unit 3.07, Level 3, KL Gateway Mall Gateway Mall, No 2, Jalan Kerinchi, Pantai Dalam, 59200 Kuala Lumpur        2
Level 12, Tower A, Plaza 33 MSC Cybercenter, 1 Jalan Kemajuan, Section 13, 46200 Petaling Jaya, Selangor, Malaysia    2
Level 21, Menara ExxonMobil, Kuala Lumpur City Centre, 50088, Kuala Lumpur                                            2
4 & 6, Lengkok Bellamy, Bukit Petaling, 50460 Kuala Lumpur, Wilayah Persekutuan Kuala Lumpur                          2
38-2, Jalan Teknologi 3/6C, Kota Damansara, 47810 Petaling Jaya, Selangor                                             2
E-05-01, First Floor\nGarden Shoppe @ One City\nJalan USJ 25/1\n47650 Subang Jaya\nSelangor, Malaysia                 2
Level 3, Menara Axis,No.2,Jalan \n51a/223,Seksyen 51a Petaling Jaya \n46100 Malaysia.         

In [25]:
df.head()

Company Name  \
0                                          Quickdesk   
1  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...   
2                                      Public Mutual   
3                    RedHill Travel & Tours Sdn Bhd    
4                   RESOURCELEAP CONSULTANCY SDN BHD   

                                     Company Address  
0  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
1  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
2  Menara Public Bank 2, No. 78, Jalan Raja Chula...  
3   located at No 36, Tingkat, 1, Jalan Ibrahim, ...  
4  A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...

In [31]:
df_new = df
# add an empty columns
df_new['Company Tel'] = ''
df_new['Company Email'] = ''
df_new['Contact Person'] = ''
df_new['Position'] = ''
df_new['FCI'] = ''
df_new['FOE'] = ''
df_new['FET'] = ''
df_new['FIST'] = ''
df_new['FOM'] = ''
df_new['FOB'] = ''
df_new['FCA'] = ''
df_new['FCM'] = ''

# print(df_new)
df_new

Company Name  \
0                           Quickdesk   
1                           QuickDesk   
2                       Public Mutual   
3     RedHill Travel & Tours Sdn Bhd    
4    RESOURCELEAP CONSULTANCY SDN BHD   
..                                ...   
285         Petronas Digital Sdn Bhd    
286          PETRONAS Digital Sdn Bhd   
287                  Petronas Digital   
288          Petronas Digital Sdn Bhd   
289              CodeHamster Solution   

                                       Company Address Company Tel  \
0    WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...               
1    WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...               
2    Menara Public Bank 2, No. 78, Jalan Raja Chula...               
3     located at No 36, Tingkat, 1, Jalan Ibrahim, ...               
4    A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...               
..                                                 ...         ...   
285  Level 21, Menara ExxonMobil, Kuala Lumpur City...               
286  Level 21, Menara ExxonMobil, Kuala Lumpur City...               
287  Level 65, Tower 2, PETRONAS Twin Towers,\nKual...               
288  Level 27, Menara Exxonmobil, Kuala Lumpur City...               
289  No. 1702, Jalan Senai Utama 1, Taman Senai Uta...               

    Company Email Contact Person Position FCI FOE FET FIST FOM FOB FCA FCM  
0                                                                           
1                                                                           
2                                                                           
3                                                                           
4                                                                           
..            ...            ...      ...  ..  ..  ..  ...  ..  ..  ..  ..  
285                                                                         
286                                                                         
287                                                                         
288                                                                         
289                                                                         

[290 rows x 14 columns]

In [30]:
df

Company Name  \
0                           Quickdesk   
1                           QuickDesk   
2                       Public Mutual   
3     RedHill Travel & Tours Sdn Bhd    
4    RESOURCELEAP CONSULTANCY SDN BHD   
..                                ...   
285         Petronas Digital Sdn Bhd    
286          PETRONAS Digital Sdn Bhd   
287                  Petronas Digital   
288          Petronas Digital Sdn Bhd   
289              CodeHamster Solution   

                                       Company Address  
0    WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
1    WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
2    Menara Public Bank 2, No. 78, Jalan Raja Chula...  
3     located at No 36, Tingkat, 1, Jalan Ibrahim, ...  
4    A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...  
..                                                 ...  
285  Level 21, Menara ExxonMobil, Kuala Lumpur City...  
286  Level 21, Menara ExxonMobil, Kuala Lumpur City...  
287  Level 65, Tower 2, PETRONAS Twin Towers,\nKual...  
288  Level 27, Menara Exxonmobil, Kuala Lumpur City...  
289  No. 1702, Jalan Senai Utama 1, Taman Senai Uta...  

[290 rows x 2 columns]

What we are trying to do, is with 9 excel spreadsheets (8 from faculties, and 1 from jobstreet), we are tryinng to merge them all into a singular master spreadsheet. This should be cleaned and standardized. 

Getting jupyter, 



In [34]:
pip install recordlinkage


     -------------------------------------- 926.9/926.9 kB 5.8 MB/s eta 0:00:00
     ------------------------------------- 206.5/206.5 kB 12.3 MB/s eta 0:00:00
Note: you may need to restart the kernel to use updated packages.



[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [35]:
import os
import pandas as pd
import recordlinkage

# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)

    # Create an indexer for record linkage
    indexer = recordlinkage.Index()
    candidate_links = indexer.full()

    # Create a comparison object
    compare_cl = recordlinkage.Compare()

    # Add comparison functions for the company name column
    compare_cl.string('Company Name', 'Company Name', method='jarowinkler', threshold=0.85)

    # Compute the potential matches
    potential_matches = compare_cl.compute(candidate_links, df)

    # Use the potential matches to create a deduplication key
    dedup_key = potential_matches.sum(axis=1) > 0

    # Deduplicate the DataFrame based on the deduplication key
    deduplicated_df = df.loc[dedup_key]

    # Print the total number of unique companies after deduplication
    unique_companies = deduplicated_df['Company Name'].nunique()
    print('Unique Companies (after deduplication):', unique_companies)

    # Find non-unique companies
    non_unique_companies = deduplicated_df['Company Name'].value_counts()[deduplicated_df['Company Name'].value_counts() > 1]

    # Print the non-unique companies
    print('Non-Unique Companies:')
    print(non_unique_companies)
else:
    print(f"File '{file_name}' does not exist in the specified folder.")


ValueError: expected pandas.MultiIndex with record pair indices as first argument

# Testing thefuzz

In [41]:
!pip install thefuzz


[notice] A new release of pip is available: 23.1.2 -> 23.2.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [45]:
import pandas as pd
from fuzzywuzzy import fuzz, process

# Creating a dataframe
dict_one = {
  "country": ["England", "Scotland", "Wales", "United Kingdom", "Northern Ireland"],
  "population_in_millions": [55.98, 5.45, 3.14, 67.33, 1.89]
}

dict_two = {
  "country": ["Northern Iland", "Wles", "Scotlnd", "Englnd", "United K."],
  "GDP_per_capita": [24900, 23882, 37460, 45101, 46510.28]
}

existing_data = pd.DataFrame(dict_one)
exported_data = pd.DataFrame(dict_two)

print(existing_data, exported_data, sep="\n\n")


            country  population_in_millions
0           England                   55.98
1          Scotland                    5.45
2             Wales                    3.14
3    United Kingdom                   67.33
4  Northern Ireland                    1.89

          country  GDP_per_capita
0  Northern Iland        24900.00
1            Wles        23882.00
2         Scotlnd        37460.00
3          Englnd        45101.00
4       United K.        46510.28


In [46]:
# Rename the misspelled columns
exported_data["country"] = exported_data["country"].apply(
  lambda x: process.extractOne(x, existing_data["country"], scorer=fuzz.partial_ratio)[0]
)
# Attempt to join the two dataframe
data = pd.merge(existing_data, exported_data, on="country", how="left")
print(data.head())

            country  population_in_millions  GDP_per_capita
0           England                   55.98        45101.00
1          Scotland                    5.45        37460.00
2             Wales                    3.14        23882.00
3    United Kingdom                   67.33        46510.28
4  Northern Ireland                    1.89        24900.00


# testing on the previous dataset

In [48]:
import os
import pandas as pd
from fuzzywuzzy import fuzz, process

# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)

    # Perform fuzzy matching to correct company names
    df["Company Name"] = df["Company Name"].apply(
        lambda x: process.extractOne(x, df["Company Name"], scorer=fuzz.partial_ratio)[0]
    )

    # Now you can proceed with deduplication or other data processing as needed
    # For example, to find non-unique companies after fuzzy matching
    non_unique_companies = df["Company Name"].value_counts()[df["Company Name"].value_counts() > 1]

    # Print the non-unique companies
    print('Non-Unique Companies:')
    print(non_unique_companies)

    # Further data processing and analysis can be done on the deduplicated DataFrame
    # For example, you can use groupby and aggregation functions to summarize data for each company.

else:
    print(f"File '{file_name}' does not exist in the specified folder.")


Non-Unique Companies:
EY                                                                                        7
Finexus International Sdn Bhd                                                             5
Telekom Malaysia Berhad                                                                   5
CTC Global Sdn Bhd                                                                        4
Fusionex                                                                                  4
Centre for Lifelong Education and Learning Innovation (LEARN) at Multimedia University    4
Petronas Digital Sdn Bhd                                                                  4
Wonder Pod Sdn Bhd                                                                        3
Infront Consulting Group (M) Sdn Bhd                                                      3
INFRONT CONSULTING                                                                        3
AIME Healthcare Sdn Bhd                                   

In [51]:
df = pd.read_excel("FCI ITP.xlsx")
df.head()

Company Name  \
0                         Quickdesk   
1                         QuickDesk   
2                     Public Mutual   
3   RedHill Travel & Tours Sdn Bhd    
4  RESOURCELEAP CONSULTANCY SDN BHD   

                                     Company Address  
0  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
1  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
2  Menara Public Bank 2, No. 78, Jalan Raja Chula...  
3   located at No 36, Tingkat, 1, Jalan Ibrahim, ...  
4  A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...

In [92]:
import os
import pandas as pd
from fuzzywuzzy import fuzz, process
import re

def preprocess_name(name):
    # Convert to lowercase and remove punctuation
    name = re.sub(r'[^\w\s]', '', name.lower())
    # Add more preprocessing steps if needed
    return name

# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)

    # Preprocess company names
    df["Company Name"] = df["Company Name"].apply(preprocess_name)

    # Perform fuzzy matching to correct company names
    df["Company Name"] = df["Company Name"].apply(
        lambda x: process.extractOne(x, df["Company Name"], scorer=fuzz.token_sort_ratio)[0]
    )

    # Now you can proceed with deduplication or other data processing as needed
    # For example, to find non-unique companies after fuzzy matching
    non_unique_companies = df["Company Name"].value_counts()[df["Company Name"].value_counts() > 1]

    # Print the non-unique companies
    print('Non-Unique Companies:')
    print(non_unique_companies)

    # Further data processing and analysis can be done on the deduplicated DataFrame
    # For example, you can use groupby and aggregation functions to summarize data for each company.

else:
    print(f"File '{file_name}' does not exist in the specified folder.")


Non-Unique Companies:
finexus international sdn bhd                                  4
ctc global sdn bhd                                             3
kollect systems sdn bhd                                        3
infront consulting group m sdn bhd                             3
ifast service centre sdn bhd                                   3
finexus group                                                  3
wonder pod sdn bhd                                             3
alnoor international school                                    3
infront consulting malaysia                                    3
convene malaysia sdn bhd                                       3
aime healthcare sdn bhd                                        3
fusionex                                                       3
petronas digital sdn bhd                                       3
ainqa health sdn bhd                                           2
experior labs sdn bhd                                          2
gtr

In [91]:
df.head()

company_id                              company_name  \
0      439382  Keysight Technologies Malaysia Sdn. Bhd.   
1      610011                      Celcom Axiata Berhad   
2      610110          Malayan Banking Berhad (Maybank)   
3      610071                                   AirAsia   
4     1319304      Petroliam Nasional Berhad (PETRONAS)   

                                             address  \
0  Bayan Lepas Free Industrial Zone, 11900 Penang...   
1  Celcom Axiata New Corporate  Office (CANCO), P...   
2  Menara Maybank Jalan Tun Perak Kuala Lumpur Wi...   
3  RedQ, Jalan Pekeliling 5, KLIA2,64000 Sepang, ...   
4  Level 3, Menara Dayabumi\r\nJalan Sultan Hisha...   

                            company_name_slug company_no  reviews_count  \
0       keysight-technologies-malaysia-439382   463532-M            884   
1                 celcom-axiata-berhad-610011   167469-A            874   
2       malayan-banking-berhad-maybank-610110     3813-K            810   
3                              airasia-610071   284669-W            727   
4  petroliam-nasional-berhad-petronas-1319304    20076-k            704   

   average_company_rating                  company_review_summary_url  \
0                  4.6041   /companies/439382/company-reviews/summary   
1                  4.4142   /companies/610011/company-reviews/summary   
2                  4.2790   /companies/610110/company-reviews/summary   
3                  4.3315   /companies/610071/company-reviews/summary   
4                  4.5384  /companies/1319304/company-reviews/summary   

                   company_review_url  foreign_company_id  \
0   /companies/439382/company-reviews                12.0   
1   /companies/610011/company-reviews              2659.0   
2   /companies/610110/company-reviews              3220.0   
3   /companies/610071/company-reviews              2973.0   
4  /companies/1319304/company-reviews            167488.0   

                                         map_address  map_latitude  \
0  Bayan Lepas Free Industrial Zone, 11900 Penang...           NaN   
1  Celcom Axiata New Corporate  Office (CANCO), P...           NaN   
2  Menara Maybank Jalan Tun Perak Kuala Lumpur Wi...           NaN   
3  RedQ, Jalan Pekeliling 5, KLIA2,64000 Sepang, ...           NaN   
4  Level 3, Menara Dayabumi\r\nJalan Sultan Hisha...           NaN   

   map_longitude                    industry                       website_url  
0            NaN    Electrical & Electronics      http://www.jobs.keysight.com  
1            NaN           Telecommunication              http://celcom.com.my  
2            NaN  Banking/Financial Services            http://www.maybank.com  
3            NaN  Aerospace/Aviation/Airline           http://www.airasia.com/  
4            NaN           Oil/Gas/Petroleum  https://www.petronas.com/join-us

In [93]:
# Extract the first name up to the first space
df['First Name'] = df['Company Name'].str.split(' ').str[0]

# Group the rows based on the 'First Name' column
grouped_df = df.groupby('First Name').agg({
    'Company Name': 'first',
    'Company Address': 'first'
}).reset_index()

print(grouped_df)

    First Name                      Company Name  \
0          act  act technology solutions sdn bhd   
1        agtiv                  agtiv consulting   
2          aic         aic semiconductor sdn bhd   
3         aime           aime healthcare sdn bhd   
4        ainqa              ainqa health sdn bhd   
..         ...                               ...   
199     wonder                wonder pod sdn bhd   
200  worldwide        worldwide holdings berhad    
201    xendity                   xendity sdn bhd   
202     zalora                            zalora   
203  zealotech      zealotech solution m sdn bhd   

                                       Company Address  
0    Lot 20-05 Level 20 Centro, 8 , Jalan Batu Tiga...  
1    D-35-05, Menara Suezcap 1, KL Gateway, Jalan K...  
2    Lot 26 & 27, Jalan Hi-tech 4, Kulim Hi-tech Pa...  
3    A-18-01 Northpoint (TowerA), Midvalley City, N...  
4    Lot 7.01 B & C, Menara BRDB,285 Jalan Maarof, ...  
..                               

In [57]:
# Save the DataFrame to a CSV file
output_file = 'grouped_companies.csv'
grouped_df.to_csv(output_file, index=False)

print(f"Data has been saved to '{output_file}'.")

Data has been saved to 'grouped_companies.csv'.


In [84]:
# Extract the first name up to the first space
df['First Name'] = df['Company Name'].str.split(' ').str[0]

# List of words to exclude from grouping
excluded_words = ['bank', 'petronas']

# Group the rows based on the 'First Name' column if the first word is not in the excluded_words list
grouped_df = df.groupby('First Name').agg({
    'Company Name': 'first',
    'Company Address': 'first'
}).reset_index()

grouped_df = grouped_df[~grouped_df['First Name'].isin(excluded_words)]

# print(grouped_df)

# Save the DataFrame to a CSV file
output_file = 'grouped_companies.csv'
grouped_df.to_csv(output_file, index=False)

print(f"Data has been saved to '{output_file}'.")

Data has been saved to 'grouped_companies.csv'.


In [123]:
# import pandas as pd

# Replace 'csv_file_path' with the actual path to your CSV file
csv_file_path = '2.jobstreet_company_info_main_industry_class.csv'

# Replace 'keyword' with the keyword you want to search for
keyword = 'publicis groupe'

# Read the CSV file into a pandas DataFrame
df = pd.read_csv(csv_file_path)

# Filter rows containing the keyword in the 'company_name' column
filtered_df = df[df['company_name'].str.contains(keyword, case=False, na=False)]

# Print the filtered DataFrame
print(filtered_df['company_name'])
filtered_df.count()


1649    Publicis Groupe Malaysia
Name: company_name, dtype: object


company_id                    1
company_name                  1
address                       1
company_name_slug             1
company_no                    1
reviews_count                 1
average_company_rating        1
company_review_summary_url    1
company_review_url            1
foreign_company_id            1
map_address                   1
map_latitude                  0
map_longitude                 0
industry                      1
website_url                   0
dtype: int64

# for now just filter the ones we have, except petronas and banks


In [100]:
# from the start
import os
import pandas as pd
# from fuzzywuzzy import fuzz, process
import re

In [105]:
# Replace 'folder_path' with the actual path to your 'ITP companies' folder
folder_path = 'ITP companies'

# Replace 'file_name' with the actual name of the Excel file
file_name = 'FCI ITP.xlsx'

# Combine folder path and file name to get the complete file path
file_path = os.path.join(folder_path, file_name)

# Check if the file exists
if os.path.exists(file_path):
    # Read the Excel file into a pandas DataFrame
    df = pd.read_excel(file_path)
#     print(df)
else:
    print(f"File '{file_name}' does not exist in the specified folder.")
    
df.head()    

Company Name  \
0                         Quickdesk   
1                         QuickDesk   
2                     Public Mutual   
3   RedHill Travel & Tours Sdn Bhd    
4  RESOURCELEAP CONSULTANCY SDN BHD   

                                     Company Address  
0  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
1  WorQ, Unit 3.07, Level 3, KL Gateway Mall Gate...  
2  Menara Public Bank 2, No. 78, Jalan Raja Chula...  
3   located at No 36, Tingkat, 1, Jalan Ibrahim, ...  
4  A-1-38, IOI Boulevard, Jalan Kenari 5, Bandar ...

In [106]:
#nonw make it lower case and add a new column named first name
def preprocess_name(name):
    # Convert to lowercase and remove punctuation
    name = re.sub(r'[^\w\s]', '', name.lower())
    # Add more preprocessing steps if needed
    return name

df_lower = df

# Preprocess company names
df_lower["Company Name"] = df_lower["Company Name"].apply(preprocess_name)

# List of words to exclude from grouping
exclude_words = ["bank", 'petronas']

# Extract the first name up to the first space
df_lower['First Name'] = df_lower['Company Name'].str.split(' ').str[0]

# Filter rows to exclude specified words
df_filtered = df_lower[~df_lower['First Name'].isin(exclude_words)]

# Group the filtered rows based on the 'First Name' column
grouped_df = df_filtered.groupby('First Name').agg({
    'Company Name': 'first',
    'Company Address': 'first'
}).reset_index()

# Save the DataFrame to a CSV file
output_file = 'grouped_companies.csv'
grouped_df.to_csv(output_file, index=False)

print(f"Data has been saved to '{output_file}'.")



# # Preprocess company names
# df_lower["Company Name"] = df_lower["Company Name"].apply(preprocess_name)

# # Perform fuzzy matching to correct company names
# df["Company Name"] = df["Company Name"].apply(
#     lambda x: process.extractOne(x, df["Company Name"], scorer=fuzz.token_sort_ratio)[0]
# )

# # Now you can proceed with deduplication or other data processing as needed
# # For example, to find non-unique companies after fuzzy matching
# non_unique_companies = df["Company Name"].value_counts()[df["Company Name"].value_counts() > 1]

# # Print the non-unique companies
# print('Non-Unique Companies:')
# print(non_unique_companies)

# # Further data processing and analysis can be done on the deduplicated DataFrame
# # For example, you can use groupby and aggregation functions to summarize data for each company.

Data has been saved to 'grouped_companies.csv'.


In [108]:
print(grouped_df)

    First Name                      Company Name  \
0          act  act technology solutions sdn bhd   
1        agtiv                  agtiv consulting   
2          aic         aic semiconductor sdn bhd   
3         aime           aime healthcare sdn bhd   
4        ainqa              ainqa health sdn bhd   
..         ...                               ...   
197     wonder                wonder pod sdn bhd   
198  worldwide        worldwide holdings berhad    
199    xendity                   xendity sdn bhd   
200     zalora                            zalora   
201  zealotech      zealotech solution m sdn bhd   

                                       Company Address  
0    Lot 20-05 Level 20 Centro, 8 , Jalan Batu Tiga...  
1    D-35-05, Menara Suezcap 1, KL Gateway, Jalan K...  
2    Lot 26 & 27, Jalan Hi-tech 4, Kulim Hi-tech Pa...  
3    A-18-01 Northpoint (TowerA), Midvalley City, N...  
4    Lot 7.01 B & C, Menara BRDB,285 Jalan Maarof, ...  
..                               

In [103]:
#works fine, they ignore bank and petronas so now i have to manually add
# import pandas as pd

df_2 = df 

# Assuming you have a DataFrame named 'df' with a column named 'Company Name'
# For example:
# df = pd.DataFrame({'Company Name': ['Petronas Holdings', 'Bank of America', 'Petroleum Corp', 'Citi Bank']})

# Convert 'Company Name' column to lowercase for case-insensitive comparison
df_2['Company Name'] = df_2['Company Name'].str.lower()

# Extract rows with first name containing 'petronas' or 'bank'
selected_rows = df_2[df_2['Company Name'].str.contains(r'^(petronas|bank)', regex=True)]

print(selected_rows)


                                  Company Name  \
52                        bank negara malaysia   
53                               bank aljazira   
88   petronas lubricants international sdn bhd   
285                  petronas digital sdn bhd    
286                   petronas digital sdn bhd   
287                           petronas digital   
288                   petronas digital sdn bhd   

                                       Company Address First Name  
52   Bank Negara Malaysia Jalan Dato Onn 50480 Kual...       bank  
53   Jeddah 23513, al Malik Abdulaziz st. Ash Shati...       bank  
88   PETRONAS Lubricants International, Melaka Lube...   petronas  
285  Level 21, Menara ExxonMobil, Kuala Lumpur City...   petronas  
286  Level 21, Menara ExxonMobil, Kuala Lumpur City...   petronas  
287  Level 65, Tower 2, PETRONAS Twin Towers,\nKual...   petronas  
288  Level 27, Menara Exxonmobil, Kuala Lumpur City...   petronas  


C:\Users\AIMAN\AppData\Local\Temp\ipykernel_20684\3032487688.py:14: UserWarning: This pattern is interpreted as a regular expression, and has match groups. To actually get the groups, use str.extract.
  selected_rows = df_2[df_2['Company Name'].str.contains(r'^(petronas|bank)', regex=True)]


In [104]:
# import pandas as pd

# Assuming you have a DataFrame named 'df' with a column named 'Company Name'
# For example:
# df = pd.DataFrame({'Company Name': ['Bank Negara Malaysia', 'Bank AlJazira', 'Petronas Lubricants International Sdn Bhd', 'Petronas Digital Sdn Bhd', 'Petronas Digital Sdn Bhd', 'Petronas Digital', 'Petronas Digital Sdn Bhd']})

# Convert 'Company Name' column to lowercase for case-insensitive comparison
selected_rows['Company Name'] = selected_rows['Company Name'].str.lower()

# Filter rows with 'Petronas Lubricants International Sdn Bhd' or containing the word 'bank'
selected_rows = selected_rows[selected_rows['Company Name'].isin(['petronas lubricants international sdn bhd']) | selected_rows['Company Name'].str.contains(r'\bbank\b')]

print(selected_rows)


                                 Company Name  \
52                       bank negara malaysia   
53                              bank aljazira   
88  petronas lubricants international sdn bhd   

                                      Company Address First Name  
52  Bank Negara Malaysia Jalan Dato Onn 50480 Kual...       bank  
53  Jeddah 23513, al Malik Abdulaziz st. Ash Shati...       bank  
88  PETRONAS Lubricants International, Melaka Lube...   petronas  


C:\Users\AIMAN\AppData\Local\Temp\ipykernel_20684\1843131361.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selected_rows['Company Name'] = selected_rows['Company Name'].str.lower()
